src/components/employee/employee-history/employee-history-component.ts

```diff
@@ -0,0 +1,19 @@
+import {Component} from 'angular2/core';
+import {AppliedLeave} from '../../../domains/applied-leave';
+import {LeaveService} from '../../../services/leave-service';
+
+@Component({
+  selector: 'employee-history',
+  templateUrl : `employee-history.tpl.html`,
+  providers: [LeaveService]
+})
+export class EmployeeHistoryComponent{
+  private appliedLeave:Array<AppliedLeave>;
+  
+  constructor(private leaveService:LeaveService){
+    
+  }
+   ngOnInit(){
+    this.leaveService.getEmployeeLeavesHistory(21).subscribe(data => this.appliedLeave = data);
```

Comment:
Please use EmployeeService's getLoggedInEmployeeId() method instead of hardcoded id.